src/Enzo/enzo_EnzoMethodMHDVlct.cpp

```diff
@@ -0,0 +1,1146 @@
+    // fields used to accumulate the changes to the conserved forms of the
+    // integrable quantities and passively advected scalars (i.e. at the start
+    // of the (partial) timestep, the fields are all set to zero and are used
+    // to accumulate the flux divergence and source terms). If CT is used, it
+    // won't have space to store changes in the magnetic fields.
+    Grouping dUcons_group;
+
+    // edge-centered electric fields
+    Grouping efield_group;
+
+    // Name of the temporary field to store the central E-field (can reuse it
+    // to store different components of the field at different times)
+    std::string center_efield_name;
+
+    // face-centered weight fields - Track the x/y/z upwind direction
+    Grouping weight_group;
+
+    // temp primitive group for storing values at the half time-step
+    Grouping temp_primitive_group;
+
+    // temp interface b-fileds
+    Grouping temp_bfieldi_group;
+
+    // allocate the temporary fields (as necessary) and fill the field groupings
+    allocate_temp_fields_(block, priml_group, primr_group,
+			  pressure_name_l, pressure_name_r,
+			  interface_velocity_name,
+			  xflux_group, yflux_group, zflux_group, dUcons_group,
+			  efield_group, center_efield_name, weight_group,
+			  temp_primitive_group, temp_bfieldi_group);
+
+    // conserved-form of passively advected scalars (the initial values are
+    // taken from here and converted to specific form. Intermediate and final
+    // values get written to here)
+    Grouping *conserved_passive_scalars = cello::field_descr()->groups();
+
+    // allocate constrained transport object
+    EnzoConstrainedTransport ct = EnzoConstrainedTransport();
+
+    double dt = block->dt();
+
+    // stale_depth indicates the number of field entries from the outermost
+    // field value that the region including "stale" values (need to be
+    // refreshed) extends over.
+    int stale_depth = 0;
+
+    // convert the passive scalars from conserved form to specific form
+    // (outside the integrator, they are treated like conserved densities)
+    compute_specific_passive_scalars_(block, passive_group_names_,
+				      *conserved_passive_scalars,
+				      *primitive_group_, stale_depth);
+
+    // repeat the following loop twice (for half time-step and full time-step)
+
+    for (int i=0;i<2;i++){
+      double cur_dt;
+      // For the purposes of making the calculation procedure slightly more
+      // more explicit, we distinguish between cur_integrable_group and
+      // cur_reconstructable_group. Due to the high level of overlap between
+      // these, they are simply aliases of the same underlying Grouping that
+      // holds groups for both of them
+      Grouping* cur_reconstructable_group;
+      Grouping* cur_integrable_group;
+      Grouping* out_integrable_group;
+
+      Grouping* cur_bfieldi_group;
+      Grouping* out_bfieldi_group;
+      EnzoReconstructor *reconstructor;
+
+      if (i == 0){
+	cur_dt = dt/2.;
```

Comment:
This is a tab vs space problem

For whatever reason my text editor (emacs) inserts tabs whenever there's an indentation with more than 8 spaces (I guess GitHub is assuming that a tab = 6 spaces?).

I seem to recall this being a common issue in other parts of the code, I'll take a look around. If this isn't common, I'll untabify the whole document. But if it isn't I'll defer this to the future (I would like to avoid doing this if possible to avoid introducing conflicts with my separate branch - but I do prefer having spaces to tabs, anyways).